### PR TITLE
Don't show "new notebook" button on pages other than the current user's own

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

File: src/server/pages/user-page.jsx [hunk 1]
```diff
@@ -50,6 +50,8 @@ export const isLoggedIn = userInfo => "name" in userInfo;
 export default class UserPage extends React.Component {
   render() {
     const { thisUser, userInfo, notebookList } = this.props;
+    const isUserAccount =
+      isLoggedIn(userInfo) && thisUser.name === userInfo.name;
```
Comment (robhudson) [inline src/server/pages/user-page.jsx]: Just out of curiosity, the `name` must be unique but what is the underlying data? Where does it come from?

Comment (wlach) [reply src/server/pages/user-page.jsx]: So we use a kind of hacky strategy on the server of embedding this kind of metadata inside the html as a script element. Here's where it comes from in this case:

https://github.com/iodide-project/iodide/blob/0e06560240c21a6665edccee496265408f33e47c/server/views.py#L71

Comment (bcolloran) [reply src/server/pages/user-page.jsx]: side note: i think we should change that hacky implementation detail at some point -- https://github.com/iodide-project/iodide/issues/1416

File: src/server/pages/user-page.jsx [hunk 2]
```diff
@@ -50,6 +50,8 @@ export const isLoggedIn = userInfo => "name" in userInfo;
 export default class UserPage extends React.Component {
   render() {
     const { thisUser, userInfo, notebookList } = this.props;
```
Comment (hamilton) [inline src/server/pages/user-page.jsx]: I've been very sloppy with these server components as I've added them, including a bit of logic in the render step, but hope we can apply the same code quality on these pages as we move forward. Is there any chance we can use this as an excuse to pull the logic out of `render` and put it in a function?

Comment (wlach) [reply src/server/pages/user-page.jsx]: Sure, what did you have in mind? A class-specific helper function or... ?

Comment (wlach) [reply src/server/pages/user-page.jsx]: As discussed, let's just merge this as-is. We can refactor this later.

**Pull request merged**